src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,104 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    public static final int CAPACITY_EMPTY_ELEMENT_DATA = 0;
```

Comment:
The constant 'CAPACITY_EMPTY_ELEMENT_DATA' is redundant since its value is zero, which is quite intuitive for an empty state. You can directly use '0' in the 'isEmpty' method.

Reply:
The constant 'CAPACITY_EMPTY_ELEMENT_DATA' is redundant since its value is zero, which is quite intuitive for an empty state. You can directly use '0' in the 'isEmpty' method.